geospaas_processing/cli/download.py

```diff
@@ -0,0 +1,78 @@
+"""
+Download files that are selected from the database using input criteria.
+"""
+import argparse
+import os
+import time
+from datetime import datetime
+import json
+import django
+
+from dateutil.relativedelta import relativedelta
+from dateutil.tz import tzutc
+from django.contrib.gis.geos import GEOSGeometry
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_processing.settings')
+django.setup()
+from geospaas.catalog.models import Dataset
+import geospaas_processing.downloaders as downloaders
+
+
+def main(arg):
+    cumulative_query=json.loads(arg.query) if arg.query else {}
+    if arg.rel_time_flag:
+        designated_begin = datetime.now().replace(tzinfo=tzutc()) + relativedelta(
+            hours=-abs(int(arg.begin)))
+        designated_end = datetime.now().replace(tzinfo=tzutc())
+    else:
+        designated_begin = datetime.strptime(arg.begin, "%Y-%m-%d").replace(tzinfo=tzutc())
+        designated_end = datetime.strptime(arg.end, "%Y-%m-%d").replace(tzinfo=tzutc())
+    download_manager = downloaders.DownloadManager(
+        download_directory=arg.down_dir.rstrip(os.path.sep),
+        provider_settings_path=arg.config_file,
+        max_downloads=int(arg.number_per_day)*(((designated_end-designated_begin).days)+1),
+        use_file_prefix=arg.use_filename_prefix,
+        time_coverage_start__gte=designated_begin,
+        time_coverage_end__lte=designated_end,
+        geographic_location__geometry__intersects=GEOSGeometry(arg.geometry),
+        **cumulative_query
+    )
+    download_manager.download()
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Process the arguments of entry_point')
+    parser.add_argument('-d', '--down_dir', required=True, type=str,
+    help="Absolute path for downloading files. If path depends on file date, usage of %Y, %m and "
+    "other placeholders interpretable by strftime is accepted")
+    parser.add_argument('-b', '--begin', required=True, type=str,
+    help="Absolute starting date for download in the format YYYY-MM-DD or (if used together "
+    "with '-r') lag in days relative to today")
+    parser.add_argument('-e', '--end', required=True, type=str,
+    help="Absolute ending date for download in the format YYYY-MM-DD or (if used together "
+    "with '-r') has no influence.")
+    parser.add_argument('-r', '--rel_time_flag', required=False, action='store_true',
+    help="The flag that distinguishes between the two cases of time calcation (1.time-lag from now "
+    "2.Two different points in time) based on its ABSENCE or PRESENCE in the arguments.")
+    parser.add_argument('-n', '--number_per_day', required=False, type=str, default="400",
+    help="limiting number of datasets that are going to be downloaded per day")
```

Comment:
So instead of 
`limiting number of datasets that are going to....`
is it better to say
`The upper limit of number of datasets that are going to...`
in this help?

Reply:
That's not the problem. It is not a limit per day. It's a safety limit on the number of datasets which can be selected by the given criteria. If the number of datasets selected by the criteria is greater than this limit, an exception is thrown and nothing is downloaded.